eslint.config.js

```diff
@@ -0,0 +1,132 @@
+        },
+      },
+    },
+    plugins: {
+      'license-header': licenseHeaderPlugin,
+      n: nPlugin,
+      promise: promisePlugin,
+      import: importPlugin,
+      prettier: prettierPlugin,
+    },
+    rules: {
+      ...eslintJs.configs.recommended.rules,
+      ...prettierConfig.rules,
+      'prettier/prettier': ['error'],
+      'license-header/header': ['error', './dev-utils/license-header.js'],
+      // Restoring some config from standardjs that we want to maintain at least
+      // for now -- to assist with transition to prettier.
+      // TODO: remove these ?????
+      'no-unused-vars': [
+        // See taav for possible better 'no-unused-vars' rule.
+        'error',
+        {
+          args: 'none',
+          caughtErrors: 'none',
+          ignoreRestSiblings: true,
+          vars: 'all',
+        },
+      ],
+      'no-empty': [
+        'error',
+        {
+          allowEmptyCatch: true,
+        },
+      ],
+      'no-constant-condition': [
+        'error',
+        {
+          checkLoops: false,
+        },
+      ],
+      'n/handle-callback-err': ['error', '^(err|error)$'],
+      'n/no-callback-literal': ['error'],
+      'n/no-deprecated-api': ['error'],
+      'n/no-exports-assign': ['error'],
+      'n/no-new-require': ['error'],
+      'n/no-path-concat': ['error'],
+      'n/process-exit-as-throw': ['error'],
+      'promise/param-names': ['error'],
+      // Undo this config from eslint:recommended for now (standardjs didn't have it.)
+      'require-yield': ['off'],
+      'import/export': 'error',
+      'import/first': 'error',
+      'import/no-absolute-path': [
+        'error',
+        { esmodule: true, commonjs: true, amd: false },
+      ],
+      'import/no-duplicates': 'error',
+      'import/no-named-default': 'error',
+      'import/no-webpack-loader-syntax': 'error',
+      // Some defaults have changed in v9
+      'dot-notation': ['error'],
+      'new-cap': ['error', { capIsNew: false }],
+      'no-eval': ['error', { allowIndirect: true }],
+      'no-new': ['error'],
+      yoda: ['error'],
+      'valid-typeof': ['error'],
+    },
+    ignores: [
+      // NOTE: Now ignore patters must be in glob expressions
+      '**/*.example.js', // a pattern for uncommited local dev files to avoid linting
+      '**/*.example.mjs', // a pattern for uncommited local dev files to avoid linting
+      '.nyc_output/**',
+      'build/**',
+      'node_modules/**',
+      '**/elastic-apm-node.js',
+      'examples/esbuild/dist/**',
+      'examples/typescript/dist/**',
+      // TODO: see comment below about JSX syntax
+      // 'examples/nextjs/**',
+      'examples/an-azure-function-app/**',
+      'lib/opentelemetry-bridge/opentelemetry-core-mini/**',
+      'test/babel/out.js',
+      'test/lambda/fixtures/esbuild-bundled-handler/hello.js',
+      // NOTE: seems that eslint is parsing this files even when we ignore it and the parser
```

Comment:
Haven't dig more into the details but even ignoring the files explicitly I cannot avoid the parsing of the file and therefore the error being thrown.

My proposal is to include them into the linting process since JSX syntax support is out of the box. There is one issue about the `no-unused-vars` rule but I decided to disable it since it would require to install a react plugin to fix it.